source/client/options_impl.cc

```diff
@@ -476,39 +531,67 @@ void OptionsImpl::validate() const {
       throw MalformedArgvException("Value for --concurrency should be greater then 0.");
     }
   }
-  try {
-    UriImpl uri(uri_);
-  } catch (const UriException&) {
-    throw MalformedArgvException("Invalid URI");
+  if (uri_.has_value()) {
+    try {
+      UriImpl uri(uri_.value());
+    } catch (const UriException&) {
+      throw MalformedArgvException("Invalid URI");
+    }
+    if (!multi_target_endpoints_.empty() || !multi_target_path_.empty() ||
+        multi_target_use_https_) {
+      throw MalformedArgvException("URI and --multi-target-* options cannot both be specified.");
+    }
+  } else {
+    if (multi_target_endpoints_.empty()) {
+      throw MalformedArgvException("A URI or --multi-target-* options must be specified.");
+    }
+    if (multi_target_path_.empty()) {
+      throw MalformedArgvException("--multi-target-path must be specified.");
+    }
   }
   try {
-    Envoy::MessageUtil::validate(*toCommandLineOptions(),
+    Envoy::MessageUtil::validate(*toCommandLineOptionsInternal(),
```

Comment:
I couldn't figure out how this indirect call to a virtual method (constructor -> validate -> toCommandLineOptions) is escaping clang-tidy on the master branch but not this branch. In any case, it does seem to be a real issue, and this does seem to be a solution.

Reply:
Oh; that is a good catch. It is pretty weird that this wasn't caught earlier. This does trigger a memory, earlier
we also ran into some odd and inconsistent clang-tidy complaints mentioning TCLAP. I don't think they pointed out the same problem -- they pointed to something in TCLAP instead (and Envoy had similar pre-existing suppressions in its codebase).

I'm not sure if what is resolved here is related to that, but I thought it might be worth bringing it up. At the time we couldn't figure out the underlying cause, and thought it was a benign issue related to TCLAP code. There are a couple of odd `NOLINT`s [1](https://github.com/envoyproxy/nighthawk/blob/67ef1094c5708d55b203b44a203b197d132c4540/source/client/options_impl.cc#L92) [2](https://github.com/envoyproxy/nighthawk/blob/67ef1094c5708d55b203b44a203b197d132c4540/source/client/options_impl.cc#L162) in this file to suppress those.

Reply:
I see, I will be on the lookout for such things involving TCLAP. I have no idea how master passes but this branch doesn't, but it might have to do with the order.

Is it better to keep all these `virtualAccessor() -> raw_field_name_` changes instead of add NOLINT to fix this particular warning?

Reply:
+1 as is. I think NOLINT here is no good as this actually looks like a real issue that got caught?